pkg/server/callbacks/v2/callbacks.go

```diff
@@ -0,0 +1,39 @@
+package callbacks
+
+import (
+	"context"
+
+	discovery "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+)
+
+// Callbacks is a collection of callbacks inserted into the server operation.
+// The callbacks are invoked synchronously
+type Callbacks interface {
```

Comment:
Why did you decide to unify callbacks back to single interface?
IE: why users that want to do sotw.NewServer now need to provide all methods, while sotw would not use them?

Reply:
So I unified these here again because as I was writing Incremental, I realized that I reused `OnStreamOpen` and `OnStreamClosed`, and I didn't like the duplication everywhere. If we can come up with a solution with minimal duplication I'm all for it, But they don't need to implement the method, writing something like: 
```go
func OnStreamDeltaRequest(int64, *discovery.DeltaDiscoveryRequest) error {
    return nil
}
```
would be perfectly acceptable, I've seen quiet a few projects do this already with the REST methods when using SOTW. I personally thought it made the code easier to understand and cleaned up the duplication but if we need to change this I'm open to a change.

Reply:
It's not about duplication, actually. It is about separation of concerns. The split was done to make those parts separate. Unifying part of it and making separate parts depend on single part, which is the union of all of those does not make actual sense.
I would use different methods like OnDeltaStreamOpen and OnDeltaStreamClosed, while the default implementation in server.go may reuse same callback for both delta and non-delta methods, unless delta callbacks are explicitly specified.

Reply:
Also, from go 1.14 it is allowed to have overlapping methods, but I would still go another route and allow users of this library to actually understand whether they received a delta or non-delta stream.

Reply:
I understand why the split was done. I'm a fan of having explicit DeltaStream callbacks. I'll make that change, I think providing those explicit callback methods does make more sense to the other. I do see how reusing could be confusing